### DISCUSSION TITLE
Enabling reading binary data under /var/lib/prometheus

### DISCUSSION THREAD

**Discussion by ripun** [prometheus/prometheus#15235] [Ideas]
### Proposal

Prometheus is used as monitoring tool for  scraping application metrics , we want to store these metrics in object storage in readable format  in JSON form so how can we do it ? but  we see this data is stored in binary format under /var/lib/prometheus

**Comment by dswarbrick** [comment]
Have you considered implementing a remote write adapter? Try looking at https://github.com/prometheus/prometheus/tree/main/documentation/examples/remote_storage/example_write_adapter

Beware that storing time series data such as Prometheus' in JSON will be a lot less efficient than the binary format. If you merely want to store data in object storage (e.g. S3), take a look at Thanos.

(this should probably be moved to a discussion too...)

**Comment by alexgreenbank** [comment]
Hello from the bug scrub.

Moving to a discussion as suggested...
